testsuite/mailhog.py

```diff
@@ -26,50 +28,84 @@ def __init__(self, openshift: 'OpenShiftClient',
             mailhog_routes = openshift.routes.for_service(mailhog_service_name)
             if len(mailhog_routes) == 0:
                 pytest.skip("Mailhog is misconfigured or missing")
-            self._url = \
-                "http://" + mailhog_routes[0]["spec"]["host"]
+            self._url = f"http://{mailhog_routes[0]['spec']['host']}"
         except OpenShiftPythonException:
             if fallback:
                 self._url = fallback
             else:
                 warn_and_skip("Can't find mailhog, skipping mailhog related tests")
 
+        self._searched_messages_ids: List[str] = []
+        self._session = requests.Session()
+
     @property
     def url(self) -> str:
         """Url of the mailhog app"""
         return self._url
 
-    def request(self, method: str = "GET", params=None, endpoint: str = None):
+    def request(self, method: str = "GET", params: dict = None, endpoint: str = None):
         """Requests the mailhog API"""
         params = params or {}
 
-        full_url = self._url + "/" + endpoint
-        response = requests.request(method=method, url=full_url,
-                                    params=params, verify=False)
+        if not hasattr(self, "_session"):
```

Comment:
Why is this condition needed? Isn´t the session always created in `__init__`?

Reply:
Should be, but from time to time it fails during test run (mostly during long debugging) and this ensures there will be always